crates/starknet/src/starknet/mod.rs

```diff
@@ -205,8 +212,8 @@ impl Starknet {
     pub(crate) fn handle_successful_transaction(
         &mut self,
         transaction_hash: &TransactionHash,
-        transaction: Transaction,
-        tx_info: TransactionExecutionInfo,
+        transaction: &TransactionWithType,
```

Comment:
Why the type is changed to TransactionWithType, there is no need for it. TransactionWithType contains the Transaction - enum and also another enum. We have the information what the transaction is based on its variant

Reply:
It was a `starknate`'s crate Transaction type. All of them were removed and now we use only ones in `types` crate

Reply:
I see that `Transaction` type is moved to types. So there is no reason to use TransactionWithType instead of Transaction, because `TransactionWithType` contains `Transaction` and its the main source of data that we are interested on.

Reply:
It was done in order not to do any conversions and just return what is stored. Anyway this type was removed [here](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/148/commits/0701750cf96d345f7df9f46596a9c420f380b65d) in [this pr](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/148)

Reply:
shall be done in a way to store just the `Transaction`, because the `TransactionWithType` is a wrapper with added field that doesn't store any information that can't be taken from the `Transaction` type itself. `TransactionWithType` can easily be converted from `Transaction` wherever needed.
Also I can't find where the `Transaction` is removed in the above mentioned pr. Maybe not all changes are pushed to it.

Reply:
But why store it and write conversions if we can skip this? You don’t have all the serialization data in Tansactions.

Reply:
This will be the best solution

Reply:
Here c522f0e7f6715ceabcb56db329694f421758bcb7